ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,284 @@
+	];
+
+	static readonly defaultKeymap_OtherPlatforms: KeymapItem[] = [
+		{ accelerator: 'Ctrl+N', command: 'newNote' },
+		{ accelerator: 'Ctrl+T', command: 'newTodo' },
+		{ accelerator: 'Ctrl+S', command: 'synchronize' },
+		{ accelerator: 'Ctrl+P', command: 'print' },
+		{ accelerator: 'Ctrl+Q', command: 'quit' },
+		{ accelerator: 'Ctrl+Alt+I', command: 'insertTemplate' },
+		{ accelerator: 'Ctrl+C', command: 'textCopy' },
+		{ accelerator: 'Ctrl+X', command: 'textCut' },
+		{ accelerator: 'Ctrl+V', command: 'textPaste' },
+		{ accelerator: 'Ctrl+A', command: 'textSelectAll' },
+		{ accelerator: 'Ctrl+B', command: 'textBold' },
+		{ accelerator: 'Ctrl+I', command: 'textItalic' },
+		{ accelerator: 'Ctrl+K', command: 'textLink' },
+		{ accelerator: 'Ctrl+`', command: 'textCode' },
+		{ accelerator: 'Ctrl+Shift+T', command: 'insertDateTime' },
+		{ accelerator: 'F6', command: 'focusSearch' },
+		{ accelerator: 'Ctrl+F', command: 'showLocalSearch' },
+		{ accelerator: 'Ctrl+Shift+S', command: 'focusElementSideBar' },
+		{ accelerator: 'Ctrl+Shift+L', command: 'focusElementNoteList' },
+		{ accelerator: 'Ctrl+Shift+N', command: 'focusElementNoteTitle' },
+		{ accelerator: 'Ctrl+Shift+B', command: 'focusElementNoteBody' },
+		{ accelerator: 'F10', command: 'toggleSidebar' },
+		{ accelerator: 'Ctrl+L', command: 'toggleVisiblePanes' },
+		{ accelerator: 'Ctrl+0', command: 'zoomActualSize' },
+		{ accelerator: 'Ctrl+E', command: 'startExternalEditing' },
+		{ accelerator: 'Ctrl+Alt+T', command: 'setTags' },
+		{ accelerator: 'Ctrl+,', command: 'config' },
+		{ accelerator: 'Ctrl+G', command: 'gotoAnything' },
+		{ accelerator: 'F1', command: 'help' },
+	];
+
+	static readonly defaultKeymap = shim.isMac()
+		? KeymapService.defaultKeymap_macOS
+		: KeymapService.defaultKeymap_OtherPlatforms;
+
+	constructor() {
+		super();
+		this.keymap = KeymapService.initKeymap();
+
+		const customKeymapPath = `${Setting.value('profileDir')}/keymap-desktop.json`;
+		if (fs.existsSync(customKeymapPath)) {
+			try {
+				const customKeymap = JSON.parse(fs.readFileSync(customKeymapPath, 'utf-8'));
+				this.setKeymap(customKeymap);
+			} catch (err) {
+				const msg = err.message ? err.message : '';
+				throw new Error(
+					`KeymapService: Failed to override keymap at ${customKeymapPath}!\n\n${msg}`
+				);
+			}
+		}
+	}
+
+	getAccelerator(command: string) {
+		const item = this.keymap[command];
+
+		if (item === undefined) {
+			throw new Error(`KeymapService: Requested "${command}" command is non-existent in the current keymap!`);
+		} else {
+			return item.accelerator;
+		}
+	}
+
+	setAccelerator(command: string, accelerator: string) {
+		// Run through localizeAccelerator() to detect duplicates with high accuracy
+		this.keymap[command].accelerator = KeymapService.localizeAccelerator(accelerator);
+	}
+
+	resetAccelerator(command: string) {
+		const defaultItem = KeymapService.defaultKeymap.find((item => item.command === command));
```

Comment:
what if the item is not found?

Reply:
Thanks for pointing that out. I'll add a catch block.